aep/general/0162/aep.md.j2

```diff
@@ -1,5 +1,372 @@
+          "path": {
+            "type": "string"
+          },
+          "alias_id": {
+            "type": "string"
+          }
+        },
+        "required": ["path", "alias"]
+      }
+    }
+  }
+}
+```
+
+{% endtabs %}
+
+- The request message **must** have a `path` field:
+  - The field **must** be [annotated as required](/field-behavior-documentation).
+  - The field **must** identify the [resource type](/resource-types) that it
+    references.
+- The request message **must** have a `alias` field:
+  - The field **must** be [annotated as required][aip-203].
+- If the user calls the method with an existing `alias`, the request **must**
+  succeed and the alias will be updated to refer to the provided revision. This
+  allows users to write code against a specific alias (e.g. `published`) and the
+  revision can change with no code change.
+
+### Rollback
+
+A common use case for a resource with a revision history is the ability to roll
+back to a given revision. APIs that support thie behavior **should** do so with
+a `Rollback` custom method:
+
+{% tab proto %}
+
+```proto
+rpc RollbackBook(RollbackBookRequest) returns (BookRevision) {
+  option (google.api.http) = {
+    post: "/v1/{name=publishers/*/books/*/revisions/*}:rollback"
+    body: "*"
+  };
+}
+
+message RollbackBookRequest {
+  // The revision that the book should be rolled back to.
+  string path = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.api.resource_reference) = {
+      type: "library.googleapis.com/BookRevision"
+    }];
+}
+```
+
+{% tab oas %}
+
+```json
+{
+  "paths": {
+    "/v1/publishers/{publisher}/books/{book}/revisions/{revision}:rollback": {
+      "post": {
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/RollbackBookRequest"
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "Successful rollback",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/BookRevision"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "RollbackBookRequest": {
+        "type": "object",
+        "properties": {
+          "path": {
+            "type": "string"
+          }
+        },
+        "required": ["path"]
+      }
+    }
+  }
+}
+```
+
+{% endtabs %}
+
+- The method **must** use the `POST` HTTP verb.
+- The method **should** return a resource revision.
+- The request message **must** have a `path` field, referring to the resource
+  revision whose configuration the resource should be rolled back to.
+  - The field **must** be [annotated as required][aip-203].
+  - The field **must** identify the [resource type][aip-123] that it
+    references.
+
+### Child resources
```

Comment:
maybe we can split this up? I think there's two points here: not understanding the existing guidance, and a question about a specific scenario that isn't explicit enough in the guidance.

For the former: what part is unclear? Is there a specific sentence we can make clearer, or a sentence that may have more than one meaning?

For the latter: we could (and probably should) be explicit about this, but if you're saying:

1. user creates resource Bookshelf, with subresource Book. (/bookshelf/x/book/y)
2. user creates a revision "A" of the Bookshelf, which contains the Book. (/bookshelf/x/revision/a/). The revision also stores it's subresources (/bookshelf/x/revision/a/book/y)
3. user deletes Book.

I'd hope that when user does a get on (/bookshelf/x/revision/a/book/y), the get the book back (since it's a subresource of the revision).

WDYT? We could definitely add more precise guidance on this case - though I'd like to do that as a follow-up PR.